torch/onnx/verification.py

```diff
@@ -136,136 +73,271 @@ def _run_ort(ort_sess, inputs):
     for input_name, input in kw_inputs.items():
         ort_inputs[input_name] = _to_numpy(input)
     inputs = _to_numpy(inputs)
-    ort_sess_inputs = ort_sess.get_inputs()
+    ort_session_inputs = ort_session.get_inputs()
     for i, input in enumerate(inputs):
-        if i == len(ort_sess_inputs) or ort_sess_inputs[i].name in ort_inputs:
+        if i == len(ort_session_inputs) or ort_session_inputs[i].name in ort_inputs:
             raise ValueError(
                 f"got too many positional inputs. inputs: {inputs}. kw_inputs: {kw_inputs}"
             )
-        ort_inputs[ort_sess_inputs[i].name] = input
-    ort_outs = ort_sess.run(None, ort_inputs)
+        ort_inputs[ort_session_inputs[i].name] = input
+    ort_outs = ort_session.run(None, ort_inputs)
     return _inline_flatten_list(ort_outs, [])
 
 
-def _ort_compare_with_pytorch(ort_outs, output, rtol, atol):
-    output, _ = torch.jit._flatten(output)
-    outputs = _unpack_to_numpy(output)
+def _create_ort_session(
+    model: Union[str, io.BytesIO], ort_providers: Tuple[str, ...] = _ORT_PROVIDERS
+):
+    try:
+        import onnxruntime  # type: ignore[import]
+    except ImportError:
+        raise ImportError("ONNXRuntime is required for export verification.")
+
+    if ort_providers is None:
+        ort_providers = _ORT_PROVIDERS
+
+    session_options = onnxruntime.SessionOptions()
+    # suppress ort warnings.
+    # 0:Verbose, 1:Info, 2:Warning. 3:Error, 4:Fatal. Default is 2.
+    session_options.log_severity_level = 3
+    ort_session = onnxruntime.InferenceSession(
+        model if isinstance(model, str) else model.getvalue(),
+        session_options,
+        providers=ort_providers,
+    )
+    return ort_session
+
+
+def _compare_ort_pytorch_outputs(ort_outs, pt_outs, rtol, atol):
+    pt_outs, _ = torch.jit._flatten(pt_outs)
+    pt_outs = _unpack_to_numpy(pt_outs)
+
+    assert len(pt_outs) == len(ort_outs), "number of outputs differ"
+
+    for ort_out, pt_out in zip(ort_outs, pt_outs):
+        np.testing.assert_allclose(ort_out, pt_out, rtol=rtol, atol=atol)
+
+
+def _prepare_input_for_pytorch(args, kwargs):
```

Comment:
I'd also lean towards removing the `prepare` prefix on these functions, but up to you.